trs_filer/ga4gh/trs/endpoints/registerObjects.py

```diff
@@ -0,0 +1,148 @@
+"""Controller for adding new objects."""
+
+from random import choice
+from typing import (Dict, List)
+import string
+
+from flask import (current_app, request)
+from pymongo.errors import DuplicateKeyError
+
+from trs_filer.app import logger
+
+
+class CreateToolPostObject:
+    """ Tool creation class. """
+
+    def __init__(self, request: request) -> None:
+        """ Initialise the ToolPost object creation.
+        Args:
+            request: API request object.
+        Returns:
+            None type.
+        """
+        self.db_collection = (
+            current_app.config['FOCA'].db.dbs['trsStore']
+            .collections['objects'].client
+        )
+        self.tool_data = request.json
+        self.id_charset = (
+            current_app.config['FOCA'].endpoints['tools']['id_charset']
+        )
+        self.id_length = (
+            current_app.config['FOCA'].endpoints['tools']['id_length']
+        )
+        self.host_name = current_app.config['FOCA'].server.host
+
+    def create_id(
+        self,
+        charset: str = ''.join([string.ascii_letters, string.digits]),
+        length: int = 6
+    ) -> str:
+        """Generate random string based on allowed set of characters.
+        Args:
+            charset: String of allowed characters.
+            length: Length of returned string.
+        Returns:
+            Random string of specified length and composed of defined set of
+            allowed characters.
+        """
+        return ''.join(choice(charset) for __ in range(length))
+
+    def create_tool_class(self) -> Dict:
+        """Create tool class.
+        Returns:
+            Generated tool class for the initialised tool object.
+        """
+        self.tool_data['toolclass'] = {
+            "description": "Temporary tool class.",
+            "id": "123456",
+            "name": "ToolClass"
+        }
+
+    def get_latest_meta_version(
+        self,
+        lst_versions: List[str]
+    ) -> str:
+        """Sort and give the latest version from version list.
+        Args:
+            lst_versions: List of versions of the tool.
+        Returns:
+            Latest version string from the list
+            if semantic versioning else empty string.
+        """
+        try:
+            lst_versions.sort(key=lambda s: list(map(int, s.split('.'))))
+            return lst_versions[-1]
+        except Exception as e:
```

Comment:
Would be better to catch one or more specific exceptions. Also, I would log this with `logger.warning()` and add the input list in the log message. Always write log messages as informative as possible, as they will help us make sure that everything runs well (and help us debug if it _doesn't_ work as expected)